db/transaction.go

```diff
@@ -18,6 +18,17 @@ func DebugNoTransaction(ctx context.Context, nonTxFunc func(context.Context) err
 	return nonTxFunc(ctx)
 }
 
+// IsolatedTransaction executes txFunc within a database transaction that is passed in to txFunc as tx Connection.
+// IsolatedTransaction returns all errors from txFunc or transaction commit errors happening after txFunc.
+// If parentConn is already a transaction, a brand new transaction will begin on the parent's connection.
+// Errors and panics from txFunc will rollback the transaction if parentConn was not already a transaction.
```

Comment:
Since we don't inherit the parent transaction, errors will always roll back the current transaction

Reply:
Fixed in https://github.com/domonda/go-sqldb/commit/077f82724f22715564a6c1f4883d905e1be9d5bb.